predicthq/client.py

```diff
@@ -45,6 +50,7 @@ def get_headers(self, headers):
         _headers.update(**headers)
         return _headers
 
+    @stamina.retry(on=(RateLimitError, ServerError), attempts=3)
```

Comment:
The number of attempts can be moved to the config if needed, I wasn't sure if it's worth doing this at this stage. What do you think?

Reply:
👌 👍 Let's implement it later if this becomes a requirement.

Reply:
I have a couple of concerns in regards to the current retry scheme:
- `ServerError` seems to be encapsulating all < `HTTP 400` & >= `HTTP 500`. We should probably make a difference between some of them so we don't retry for it (e.g. `HTTP 500`)
- Do you think the default jitter of `stamina` would be enough when the client will start receiving `429`s ? Or would it help to reduce the amount of errors to extend the wait time before trying a request again ?

Reply:
I agree with your concerns @lpouvreau I will take another look at how to improve the logic.

Reply:
I agree with @lpouvreau 1st item - regarding jitter I am unsure